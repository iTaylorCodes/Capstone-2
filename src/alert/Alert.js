import { v4 as uuidv4 } from "uuid";

/** Alert component
 *
 * Shows bootstrap style alerts
 *
 * Used for LoginForm, SignupForm, and Profile components
 */

const Alert = ({ type = "danger", messages = [] }) => {
  return (
    <div className={`alert alert-${type} m-3`} role="alert">
      {messages.map((error) => (
        <p className="mb-0 small" key={uuidv4()}>
          {error}
        </p>
      ))}
    </div>
  );
};

export default Alert;
