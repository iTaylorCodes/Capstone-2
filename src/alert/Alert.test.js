import { render } from "@testing-library/react";
import Alert from "./Alert";

it("renders without crashing", function () {
  render(<Alert />);
});

it("matches snapshot for danger", function () {
  let messages = ["Broken!", "Not Found"];
  const { asFragment } = render(<Alert type="danger" messages={messages} />);
  expect(asFragment()).toMatchSnapshot;
});

it("matches snapshot for success", function () {
  let messages = ["Working!"];
  const { asFragment } = render(<Alert type="success" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});
