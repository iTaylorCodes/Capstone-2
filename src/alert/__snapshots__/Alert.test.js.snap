// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success m-3"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Working!
    </p>
  </div>
</DocumentFragment>
`;
